receiver/vmmetricsreceiver/vm_metrics_collector.go

```diff
@@ -84,8 +88,26 @@ func NewVMMetricsCollector(si time.Duration, mountPoint, processMountPoint, pref
 	return vmc, nil
 }
 
+func detectResource() {
+	res, err := auto.Detect(context.Background())
+	if err != nil {
+		panic(fmt.Sprintf("Resource detection failed, err:%v", err))
+	}
+	if res != nil {
+		rsc = &resourcepb.Resource{
```

Comment:
one last q: since it is shared between multiple instances it seems reasonable to have it protected by a `sync.Once`, especially taking into account this attribution here.

Reply:
I'll add the sync.Once.